ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -59,39 +59,54 @@ let initialize_info : InitializeResult.t =
 
 let ocamlmerlin_reason = "ocamlmerlin-reason"
 
-let send_diagnostics rpc doc =
+let send_diagnostics ?diagnostics rpc doc =
   let state : State.t = Server.state rpc in
-  let send () =
```

Comment:
The old code was a little smarter.  Diagnostics were only created right before they went sent which saves us from discarding them when newer diagnostics are scheduled.

Reply:
Am I not doing the same? Have you noticed `async @@ fun () -> ...`?